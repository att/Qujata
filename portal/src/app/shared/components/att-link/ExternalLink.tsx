import { BaseLinkProps, LinkRel, LinkTarget } from './Link.model';
import { getLinkClassName } from './Link.util';

export interface ExternalLinkProps extends BaseLinkProps {
  target?: LinkTarget;
  rel?: LinkRel;
}

export const ExternalLink: React.FC<ExternalLinkProps> = (props: ExternalLinkProps) => (
  <a
    href={props.link}
    target={props.target || LinkTarget.SELF}
    rel={props.rel || LinkRel.NONE}
    className={getLinkClassName(props)}
    onClick={props.onLinkClick}
    title={props.linkTitle}
    data-cy={props.e2eId}
  >
    {props.children}
  </a>
);
