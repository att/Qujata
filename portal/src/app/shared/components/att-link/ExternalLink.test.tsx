import { render, RenderResult } from '@testing-library/react';
import { ExternalLink, ExternalLinkProps } from './ExternalLink';
import { LinkRel, LinkSize, LinkStyle, LinkTarget } from './Link.model';

describe('ExternalLink', () => {
  test('renders external wrapper link', () => {
    const props: ExternalLinkProps = {
      link: 'test-link',
      styleType: LinkStyle.WRAPPER,
      size: LinkSize.NONE,
      target: LinkTarget.SELF,
      rel: LinkRel.NO_OPENER,
      children: undefined,
    };
    const { container }: RenderResult = render(<ExternalLink {...props}><span>external wrapper link</span></ExternalLink>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders external link', () => {
    const props: ExternalLinkProps = {
      link: 'test-link',
      styleType: LinkStyle.NONE,
      size: LinkSize.NONE,
      target: LinkTarget.SELF,
      rel: LinkRel.NO_OPENER,
      className: 'test-class-name',
      children: undefined,
    };
    const { container }: RenderResult = render(<ExternalLink {...props}>external link</ExternalLink>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('test-class-name');
  });

  test('renders external link with default target and rel attributes', () => {
    const props: ExternalLinkProps = {
      link: 'test-link',
      styleType: LinkStyle.NONE,
      size: LinkSize.NONE,
      className: 'test-class-name',
      children: undefined,
    };
    const { container }: RenderResult = render(<ExternalLink {...props}>external link</ExternalLink>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
