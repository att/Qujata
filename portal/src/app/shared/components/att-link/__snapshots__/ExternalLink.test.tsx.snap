// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders external link 1`] = `
<a
  class="test-class-name"
  href="test-link"
  rel="noopener"
  target="_self"
>
  external link
</a>
`;

exports[`ExternalLink renders external link with default target and rel attributes 1`] = `
<a
  class="test-class-name"
  href="test-link"
  rel=""
  target="_self"
>
  external link
</a>
`;

exports[`ExternalLink renders external wrapper link 1`] = `
<a
  class="wrapper_link"
  href="test-link"
  rel="noopener"
  target="_self"
>
  <span>
    external wrapper link
  </span>
</a>
`;
